esp/src/src-react/components/Scopes.tsx

```diff
@@ -0,0 +1,358 @@
+import * as React from "react";
+import { CommandBar, ContextualMenuItemType, ICommandBarItemProps, Icon, Link, mergeStyleSets } from "@fluentui/react";
+import { DFUService } from "@hpcc-js/comms";
+import { SizeMe } from "react-sizeme";
+import * as WsDfu from "src/WsDfu";
+import { formatCost } from "src/Session";
+import * as Utility from "src/Utility";
+import nlsHPCC from "src/nlsHPCC";
+import { useConfirm } from "../hooks/confirm";
+import { useFluentGrid } from "../hooks/grid";
+import { useBuildInfo } from "../hooks/platform";
+import { useUserTheme } from "../hooks/theme";
+import { HolyGrail } from "../layouts/HolyGrail";
+import { pushParams } from "../util/history";
+import { AddToSuperfile } from "./forms/AddToSuperfile";
+import { CopyFile } from "./forms/CopyFile";
+import { DesprayFile } from "./forms/DesprayFile";
+import { Fields } from "./forms/Fields";
+import { Filter } from "./forms/Filter";
+import { RemoteCopy } from "./forms/RemoteCopy";
+import { RenameFile } from "./forms/RenameFile";
+import { ShortVerticalDivider } from "./Common";
+
+const FilterFields: Fields = {
+    "LogicalName": { type: "string", label: nlsHPCC.Name, placeholder: nlsHPCC.somefile },
+    "Description": { type: "string", label: nlsHPCC.Description, placeholder: nlsHPCC.SomeDescription },
+    "Owner": { type: "string", label: nlsHPCC.Owner, placeholder: nlsHPCC.jsmi },
+    "Index": { type: "checkbox", label: nlsHPCC.Index },
+    "NodeGroup": { type: "target-group", label: nlsHPCC.Group, placeholder: nlsHPCC.Cluster },
+    "FileSizeFrom": { type: "string", label: nlsHPCC.FromSizes, placeholder: "4096" },
+    "FileSizeTo": { type: "string", label: nlsHPCC.ToSizes, placeholder: "16777216" },
+    "FileType": { type: "file-type", label: nlsHPCC.FileType },
+    "FirstN": { type: "string", label: nlsHPCC.FirstN, placeholder: "-1" },
+    // "Sortby": { type: "file-sortby", label: nlsHPCC.FirstNSortBy, disabled: (params: Fields) => !params.FirstN.value },
```

Comment:
I meant to ask about this, in the dojo version the Filter button becomes disabled while viewing by scopes. Should it be removed altogether from the react version? Or we could use DFUQuery (which would allow for the filter params) instead of (the current) DFUFileView requests for populating the grid?

Reply:
Open a new Jira for this and keep it as a separate exercise?